eng/referenceFromRuntime.targets

```diff
@@ -1,5 +1,5 @@
 <Project>
-  <PropertyGroup Condition="'$(TargetsNetCoreApp)' == 'true' and '$(TargetFramework)' != '$(NetCoreAppCurrent)'">
+  <PropertyGroup Condition="$(TargetFramework.StartsWith('netcoreapp')) and '$(TargetFramework)' != '$(NetCoreAppCurrent)'">
```

Comment:
This should be TargetFrameworkIdentifier. Inside the project file you should condition only on TargetFramework but outside it when TargetFrameworkIdentifier is available you should prefer it.

Reply:
Same for all other cases in after targets.

Reply:
Would you mind elaborating why for the sake of knowing?

Reply:
Thanks for reminding me to post the references.

NuGet has a long standing issue because they parse TargetFramework(s) even though those properties are user "aliases". (https://github.com/dotnet/project-system/blob/master/docs/repo/coding-conventions.md#data): https://github.com/NuGet/Home/issues/5154.

The recommended from the SDK/project-system is to use TargetFramework only where you have control over it, ie conditions in the project file itself. Ideally you would always use `TargetFrameworkIdentifier` and `TargetFrameworkVersion` but as those properties aren't available in the project file before the Common targets are imported, that's not possible.

The obvious advantage of using `TargetFrameworkIdentifier` is that this won't break if the user "alias" (TargetFramework(s)) gets overloaded further (ie -OS part) or when the TargetFrameworkMoniker changes (ie netcoreapp5.0 -> net5.0).